### PR TITLE
HIVE-28577: Upgrade protobuf version to 3.25.5 to fix CVE

### PR DESCRIPTION
**Pull request by Aggarwal-Raghav**
### What changes were proposed in this pull request?
Upgrading protobuf version to 3.25.5

### Why are the changes needed?
To fix CVE: [CVE-2024-7254](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-7254) and to be in sync with TEZ (master branch) protobuf version as well.

### Does this PR introduce _any_ user-facing change?
NO

### Is the change a dependency upgrade?
YES
[dependency_tree.txt](https://github.com/user-attachments/files/17374184/dependency_tree.txt)

### How was this patch tested?
Will see output of Jenkins

### CODE REVIEW

**Comment by Aggarwal-Raghav** [comment]
For protobuf 3.25.5 generated code:
1. Downloaded protoc 3.25.5 and set it in the classpath.
2. Built hive project with the command: `mvn clean install -DskipTests -Pitests,dist,qsplits,errorProne,protobuf -Drat.skip=true`

**Comment by Aggarwal-Raghav** [comment]
@ayushtkn , can you please review this?

File: llap-common/src/gen/protobuf/gen-java/org/apache/hadoop/hive/llap/daemon/rpc/LlapDaemonProtocolProtos.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Generated by the protocol buffer compiler.  DO NOT EDIT!
 // source: LlapDaemonProtocol.proto
 
+// Protobuf Java Version: 3.25.5
```
Comment (difin) [inline llap-common/src/gen/protobuf/gen-java/org/apache/hadoop/hive/llap/daemon/rpc/LlapDaemonProtocolProtos.java:4]: Is this a manual change or an auto-generated?
If manual, why is it needed?

Comment (Aggarwal-Raghav) [reply llap-common/src/gen/protobuf/gen-java/org/apache/hadoop/hive/llap/daemon/rpc/LlapDaemonProtocolProtos.java:4]: This is auto-generated code. There is a `protobuf` profile that generates it.

**Review by ayushtkn — APPROVED** [review]
LGTM

**Pull request merged**